### PR TITLE
Various changes to the API draft

### PR DESCRIPTION
**Pull request by freyacodes**
This is a pr on top of #750

* Expanded the change list
* Added future changes in v4 and v5
* Moved WS handshake to /v3/websocket
* Changed structure of Track
* Removed Voice Event, replaced with existing Voice State
* Clarify various things
* And more

### CODE REVIEW

File: IMPLEMENTATION.md [hunk 1]
```diff
@@ -418,37 +433,46 @@ GET /v3/sessions/{sessionId}/players
...
+| sourceName | string  | The track source name                     |
+
+##### Voice State
+| Field     | Type   | Description                                           |
+|-----------|--------|-------------------------------------------------------|
+| token     | string | The Discord voice token to authenticate with          |
+| endpoint  | string | The Discord voice endpoint to connect to              |
+| sessionId | string | The Discord voice session id to authenticate with     |
+| connected | ?bool  | Whether the player is connected. Response only        |
+| ping      | ?int   | Roundtrip latency to the voice gateway. Response only |
```
Comment (topi314) [inline IMPLEMENTATION.md:471]: I'm not happy with the header name but this should be more clear
```suggestion
| Field     | Type   | Description                                       | Valid Types       |
|-----------|--------|---------------------------------------------------|-------------------|
| token     | string | The Discord voice token to authenticate with      | Request, Response |
| endpoint  | string | The Discord voice endpoint to connect to          | Request, Response |
| sessionId | string | The Discord voice session id to authenticate with | Request, Response |
| connected | ?bool  | Whether the player is connected                   | Request           |
| ping      | ?int   | Roundtrip latency to the voice gateway            | Request           |
```

File: IMPLEMENTATION.md [hunk 2]
```diff
@@ -993,40 +1017,23 @@ When a search prefix is used, the returned `loadType` will be `SEARCH_RESULT`. N
...
-  "title": "Rick Astley - Never Gonna Give You Up",
-  "uri": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
-  "sourceName": "youtube"
-}
-```
-</details>
-
-<details>
-<summary>Response(Deprecated)</summary>
+<summary>Response</summary>
```
Comment (topi314) [inline IMPLEMENTATION.md]:
```suggestion
<summary>Example Payload</summary>
```

File: IMPLEMENTATION.md [hunk 3]
```diff
@@ -1070,34 +1076,13 @@ Response:
...
-    "sourceName": "youtube"
-  },
-  ...
-]
-```
-</details>
-
-<details>
-<summary>Response(Deprecated)</summary>
+<summary>Response</summary>
```
Comment (topi314) [inline IMPLEMENTATION.md]:
```suggestion
<summary>Example Payload</summary>
```

**Review by topi314 — CHANGES_REQUESTED** [review]
looks good to me mostly :)

File: IMPLEMENTATION.md (hunk 1 above)
Comment (freyacodes) [reply IMPLEMENTATION.md:471]: It feels like this adds a lot more information without actually adding anything. I don't personally like this

Comment (topi314) [reply IMPLEMENTATION.md:471]: It would be consistent with the other tables in this regard and make it clearer

**Review by topi314 — APPROVED** [review]

**Pull request merged**